---
title: "indiv_coverage"
author: "Jacob Green"
date: '2023-09-29'
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(root.dir = "/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")
```

## Figure 2 capture read depth across genomic features for insert norm coverage

```{r Set Libraries and Working Directory, echo = FALSE}
library(ggplot2)
library(grid)
library(knitr)
library(dplyr)
library(scales)
library(zoo)
library(gridExtra)
library(grid)
library(ggpubr)
library(cowplot)
library(tidyverse)
```

```{r Capture_B1, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")
# initialize an empty list to store ggplot objects
plot_list <- list()
files <- c("capture1_b1", "capture4_b1", "capture7_b1", "capture10_b1", "capture13_b1", "capture16_b1")
for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)
  labs <- c("CDS","Exon","Gene","UTR")
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }
  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")
  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")
  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  plot_list[[length(plot_list)+1]] <- p1
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  print(p1)
  dev.off()
}
# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 3, top=textGrob("Capture b1 read depth across genome features"))

#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_B1.png", pcombined)
```

```{r Capture_B2, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("capture1_b2", "capture4_b2", "capture7_b2", "capture10_b2", "capture13_b2", "capture16_b2")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample)) + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    #geom_segment(aes(x=20, y=0, xend=20, yend=1, color="red"))+
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 3, top=textGrob("Capture b2 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_B2.png", pcombined)
```

```{r Capture_G1, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("capture1_g1", "capture4_g1", "capture7_g1", "capture10_g1", "capture13_g1", "capture16_g1")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top=textGrob("Capture g1 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_g1.png", pcombined)
```

```{r Capture_G7, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("capture1_g7", "capture4_g7", "capture7_g7", "capture10_g7", "capture13_g7", "capture16_g7")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top=textGrob("Capture g7 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_G7.png", pcombined)
```

```{r Capture_K3, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1a/04_coverage_analysis/01_genome_region/02_indiv")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("Capture1.K3", "capture4.K3", "Capture3.K3", "Capture4.K3")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top=textGrob("Capture K3 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_K3.png", pcombined)
```

```{r Capture_K4, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1a/04_coverage_analysis/01_genome_region/02_indiv")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("Capture1.K4", "capture4.K4", "Capture3.K4", "Capture4.K4")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top=textGrob("Capture B4 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_K4.png", pcombined)
```

```{r Capture_M3, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1a/04_coverage_analysis/01_genome_region/02_indiv")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("Capture1.M3", "capture4.M3", "Capture3.M3", "Capture4.M3")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top=textGrob("Capture M3 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_M3.png", pcombined)
```

```{r Capture_M4, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1a/04_coverage_analysis/01_genome_region/02_indiv")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("Capture1.M4", "capture4.M4", "Capture3.M4", "Capture4.M4")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top=textGrob("Capture M4 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_M4.png", pcombined)
```

```{r Capture_N2, message=FALSE, warning=TRUE, include=FALSE}
setwd("/home/jgreen/eager_obj1a/04_coverage_analysis/01_genome_region/02_indiv")
# initialize an empty list to store ggplot objects
plot_list <- list()

files <- c("Capture1.N2", "capture4.N2", "Capture3.N2", "Capture4.N2")

for (file in files) {
  print(file)
  files_list <- list.files(pattern=file)
  print(files_list)
  
  files <- c(paste0(file, ".AllCDS.all.split.txt"), paste0(file, ".AllExon.all.split.txt"), paste0(file, ".AllGene.all.split.txt"), paste0(file, ".AllUTR.all.split.txt"))
  print(files)

  labs <- c("CDS","Exon","Gene","UTR")
  
  cov <- list()
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[,c(2,5)]
    cov_cumul=1-cumsum(cov[[i]][,2])
    cov[[i]]$cov_cumul <- c(1,cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample=labs[i]
  }

  cov_df=do.call("rbind",cov)
  names(cov_df)[1:2]=c("depth","fraction")

  pcbPalette <- c("#009E73" ,"#D55E00","#CC79A7","#56B4E9")

  p1 <- ggplot(cov_df, aes(x= depth, y=cov_cumul, color=sample))  + xlim(0,5)+
    scale_alpha(guide = 'none') +
    geom_line(size=1.5)+ 
    scale_color_manual(values=pcbPalette) +
    scale_fill_manual(values=pcbPalette) +
    ggtitle(file)+
    ylab("% of Bases > Depth")+
    xlab("Depth")+
    theme_bw() +
    theme(plot.title = element_text(size = 12, face = "bold",hjust = 0.5)) +
    theme(legend.position="none")
  
  plot_list[[length(plot_list)+1]] <- p1
  
  png(filename=paste0("Figure2_", file, ".png"), type="cairo",units="px", width=5600, 
      height=3000, res=600, bg="transparent")
  
  print(p1)
  
  dev.off()

}

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top=textGrob("Capture N2 Read Depth across genome"))
#save the combined plot as PNG file
ggsave(file="Figure2_indivnorm_N2.png", pcombined)
```

```{r Capture 1, message=FALSE, warning=TRUE, include=FALSE}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the capture you want to isolate
capture <- "capture1"

# Initialize an empty list to store ggplot objects
plot_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Loop through each sample
for (sample in samples) {
  # Create the file name based on the sample and capture
  file <- paste0(capture, "_", sample)
  
  # Print the current file being processed
  print(file)
  
  # Get a list of files for the current capture and sample
  files_list <- list.files(pattern = file)
  print(files_list)
  
  # Specify the files you want to process for each sample
  files <- c(
    paste0(file, ".AllCDS.all.split.txt"),
    paste0(file, ".AllExon.all.split.txt"),
    paste0(file, ".AllGene.all.split.txt"),
    paste0(file, ".AllUTR.all.split.txt")
  )
  print(files)
  
  # Labels for different data types
  labs <- c("CDS", "Exon", "Gene", "UTR")
  
  # Initialize a list to store data for each data type
  cov <- list()
  
  # Loop through each file and data type
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[i]][, 2])
    cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample <- labs[i]
  }
  
  # Combine data for all data types
  cov_df <- do.call("rbind", cov)
  names(cov_df)[1:2] <- c("depth", "fraction")
  
  # Color palette
  pcbPalette <- c("#009E73", "#D55E00", "#CC79A7", "#56B4E9")
  
  # Create a ggplot
  p1 <- ggplot(cov_df, aes(x = depth, y = cov_cumul, color = sample, linetype = sample)) +
    xlim(0, 10) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    scale_color_manual(values = pcbPalette) +
    scale_fill_manual(values = pcbPalette) +
    ggtitle(paste("Read Depth Distribution for", capture, "(", sample, ")")) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(axis.text = element_text(size = 5),axis.title = element_text(size = 5,face = "bold")) +
    theme(plot.title = element_text(size = 5, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")
  
  # Append the plot to the list
  plot_list[[sample]] <- p1
  
  # Save the individual plot as a PNG file
  png(
    filename = paste0("Figure2_", capture, "_", sample, ".png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
}

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(
  grobs = plot_list,
  ncol = 5,  # Adjust the number of columns as needed
  top = textGrob(paste("Read Depth Distribution for", capture, "across Samples"))
)

# Save the combined plot as a PNG file
ggsave(file = paste0("Figure2_", capture, "_Combined.png"), pcombined)

```

```{r Capture 4, message=FALSE, warning=TRUE, include=FALSE}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the capture you want to isolate
capture <- "capture4"

# Initialize an empty list to store ggplot objects
plot_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Loop through each sample
for (sample in samples) {
  # Create the file name based on the sample and capture
  file <- paste0(capture, "_", sample)
  
  # Print the current file being processed
  print(file)
  
  # Get a list of files for the current capture and sample
  files_list <- list.files(pattern = file)
  print(files_list)
  
  # Specify the files you want to process for each sample
  files <- c(
    paste0(file, ".AllCDS.all.split.txt"),
    paste0(file, ".AllExon.all.split.txt"),
    paste0(file, ".AllGene.all.split.txt"),
    paste0(file, ".AllUTR.all.split.txt")
  )
  print(files)
  
  # Labels for different data types
  labs <- c("CDS", "Exon", "Gene", "UTR")
  
  # Initialize a list to store data for each data type
  cov <- list()
  
  # Loop through each file and data type
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[i]][, 2])
    cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample <- labs[i]
  }
  
  # Combine data for all data types
  cov_df <- do.call("rbind", cov)
  names(cov_df)[1:2] <- c("depth", "fraction")
  
  # Color palette
  pcbPalette <- c("#009E73", "#D55E00", "#CC79A7", "#56B4E9")
  
  # Create a ggplot
  p1 <- ggplot(cov_df, aes(x = depth, y = cov_cumul, color = sample, linetype = sample)) +
    xlim(0, 10) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    scale_color_manual(values = pcbPalette) +
    scale_fill_manual(values = pcbPalette) +
    ggtitle(paste("Read Depth Distribution for", capture, "(", sample, ")")) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(axis.text = element_text(size = 5),axis.title = element_text(size = 5,face = "bold")) +
    theme(plot.title = element_text(size = 5, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")
  
  # Append the plot to the list
  plot_list[[sample]] <- p1
  
  # Save the individual plot as a PNG file
  png(
    filename = paste0("Figure2_", capture, "_", sample, ".png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
}

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(
  grobs = plot_list,
  ncol = 5,  # Adjust the number of columns as needed
  top = textGrob(paste("Read Depth Distribution for", capture, "across Samples"))
)

# Save the combined plot as a PNG file
ggsave(file = paste0("Figure2_", capture, "_Combined.png"), pcombined)
```

```{r Capture 7, message=FALSE, warning=TRUE, include=FALSE}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the capture you want to isolate
capture <- "capture7"

# Initialize an empty list to store ggplot objects
plot_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Loop through each sample
for (sample in samples) {
  # Create the file name based on the sample and capture
  file <- paste0(capture, "_", sample)
  
  # Print the current file being processed
  print(file)
  
  # Get a list of files for the current capture and sample
  files_list <- list.files(pattern = file)
  print(files_list)
  
  # Specify the files you want to process for each sample
  files <- c(
    paste0(file, ".AllCDS.all.split.txt"),
    paste0(file, ".AllExon.all.split.txt"),
    paste0(file, ".AllGene.all.split.txt"),
    paste0(file, ".AllUTR.all.split.txt")
  )
  print(files)
  
  # Labels for different data types
  labs <- c("CDS", "Exon", "Gene", "UTR")
  
  # Initialize a list to store data for each data type
  cov <- list()
  
  # Loop through each file and data type
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[i]][, 2])
    cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample <- labs[i]
  }
  
  # Combine data for all data types
  cov_df <- do.call("rbind", cov)
  names(cov_df)[1:2] <- c("depth", "fraction")
  
  # Color palette
  pcbPalette <- c("#009E73", "#D55E00", "#CC79A7", "#56B4E9")
  
  # Create a ggplot
  p1 <- ggplot(cov_df, aes(x = depth, y = cov_cumul, color = sample, linetype = sample)) +
    xlim(0, 10) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    scale_color_manual(values = pcbPalette) +
    scale_fill_manual(values = pcbPalette) +
    ggtitle(paste("Read Depth Distribution for", capture, "(", sample, ")")) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(axis.text = element_text(size = 5),axis.title = element_text(size = 5,face = "bold")) +
    theme(plot.title = element_text(size = 5, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")
  
  # Append the plot to the list
  plot_list[[sample]] <- p1
  
  # Save the individual plot as a PNG file
  png(
    filename = paste0("Figure2_", capture, "_", sample, ".png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
}

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(
  grobs = plot_list,
  ncol = 3,  # Adjust the number of columns as needed
  top = textGrob(paste("Read Depth Distribution for", capture, "across Samples"))
)

# Save the combined plot as a PNG file
ggsave(file = paste0("Figure2_", capture, "_Combined.png"), pcombined)
```

```{r Capture 10, message=FALSE, warning=TRUE, include=FALSE}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the capture you want to isolate
capture <- "capture10"

# Initialize an empty list to store ggplot objects
plot_list <- list()

# List of all samples
samples <- c("b1", "b2", "b6", "b7", "b10", "g1", "g5", "g6", "g7", "g8", "k1", "k2", "k3", "k5", "k8", "m1", "m3", "m5", "m7", "m9", "n1", "n2", "n3", "n4", "n5")

# Loop through each sample
for (sample in samples) {
  # Create the file name based on the sample and capture
  file <- paste0(capture, "_", sample)
  
  # Print the current file being processed
  print(file)
  
  # Get a list of files for the current capture and sample
  files_list <- list.files(pattern = file)
  print(files_list)
  
  # Specify the files you want to process for each sample
  files <- c(
    paste0(file, ".AllCDS.all.split.txt"),
    paste0(file, ".AllExon.all.split.txt"),
    paste0(file, ".AllGene.all.split.txt"),
    paste0(file, ".AllUTR.all.split.txt")
  )
  print(files)
  
  # Labels for different data types
  labs <- c("CDS", "Exon", "Gene", "UTR")
  
  # Initialize a list to store data for each data type
  cov <- list()
  
  # Loop through each file and data type
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[i]][, 2])
    cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample <- labs[i]
  }
  
  # Combine data for all data types
  cov_df <- do.call("rbind", cov)
  names(cov_df)[1:2] <- c("depth", "fraction")
  
  # Color palette
  pcbPalette <- c("#009E73", "#D55E00", "#CC79A7", "#56B4E9")
  
  # Create a ggplot
  p1 <- ggplot(cov_df, aes(x = depth, y = cov_cumul, color = sample, linetype = sample)) +
    xlim(0, 10) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    scale_color_manual(values = pcbPalette) +
    scale_fill_manual(values = pcbPalette) +
    ggtitle(paste("Read Depth Distribution for", capture, "(", sample, ")")) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(axis.text = element_text(size = 5),axis.title = element_text(size = 5,face = "bold")) +
    theme(plot.title = element_text(size = 5, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")
  
  # Append the plot to the list
  plot_list[[sample]] <- p1
  
  # Save the individual plot as a PNG file
  png(
    filename = paste0("Figure2_", capture, "_", sample, ".png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
}

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(
  grobs = plot_list,
  ncol = 3,  # Adjust the number of columns as needed
  top = textGrob(paste("Read Depth Distribution for", capture, "across Samples"))
)

# Save the combined plot as a PNG file
ggsave(file = paste0("Figure2_", capture, "_Combined.png"), pcombined)
```

```{r Capture 13, message=FALSE, warning=TRUE, include=FALSE}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the capture you want to isolate
capture <- "capture13"

# Initialize an empty list to store ggplot objects
plot_list <- list()

# List of all samples
samples <- c("b1", "b2", "b6", "b7", "b10", "g1", "g5", "g6", "g7", "g8", "k1", "k2", "k3", "k5", "k8", "m1", "m3", "m5", "m7", "m9", "n1", "n2", "n3", "n4", "n5")

# Loop through each sample
for (sample in samples) {
  # Create the file name based on the sample and capture
  file <- paste0(capture, "_", sample)
  
  # Print the current file being processed
  print(file)
  
  # Get a list of files for the current capture and sample
  files_list <- list.files(pattern = file)
  print(files_list)
  
  # Specify the files you want to process for each sample
  files <- c(
    paste0(file, ".AllCDS.all.split.txt"),
    paste0(file, ".AllExon.all.split.txt"),
    paste0(file, ".AllGene.all.split.txt"),
    paste0(file, ".AllUTR.all.split.txt")
  )
  print(files)
  
  # Labels for different data types
  labs <- c("CDS", "Exon", "Gene", "UTR")
  
  # Initialize a list to store data for each data type
  cov <- list()
  
  # Loop through each file and data type
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[i]][, 2])
    cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample <- labs[i]
  }
  
  # Combine data for all data types
  cov_df <- do.call("rbind", cov)
  names(cov_df)[1:2] <- c("depth", "fraction")
  
  # Color palette
  pcbPalette <- c("#009E73", "#D55E00", "#CC79A7", "#56B4E9")
  
  # Create a ggplot
  p1 <- ggplot(cov_df, aes(x = depth, y = cov_cumul, color = sample, linetype = sample)) +
    xlim(0, 10) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    scale_color_manual(values = pcbPalette) +
    scale_fill_manual(values = pcbPalette) +
    ggtitle(paste("Read Depth Distribution for", capture, "(", sample, ")")) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(axis.text = element_text(size = 5),axis.title = element_text(size = 5,face = "bold")) +
    theme(plot.title = element_text(size = 5, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")
  
  # Append the plot to the list
  plot_list[[sample]] <- p1
  
  # Save the individual plot as a PNG file
  png(
    filename = paste0("Figure2_", capture, "_", sample, ".png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
}

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(
  grobs = plot_list,
  ncol = 3,  # Adjust the number of columns as needed
  top = textGrob(paste("Read Depth Distribution for", capture, "across Samples"))
)

# Save the combined plot as a PNG file
ggsave(file = paste0("Figure2_", capture, "_Combined.png"), pcombined)
```

```{r Capture 16, message=FALSE, warning=TRUE, include=FALSE}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the capture you want to isolate
capture <- "capture16"

# Initialize an empty list to store ggplot objects
plot_list <- list()

# List of all samples
samples <- c("b1", "b2", "b6", "b7", "b10", "g1", "g5", "g6", "g7", "g8", "k1", "k2", "k3", "k5", "k8", "m1", "m3", "m5", "m7", "m9", "n1", "n2", "n3", "n4", "n5")

# Loop through each sample
for (sample in samples) {
  # Create the file name based on the sample and capture
  file <- paste0(capture, "_", sample)
  
  # Print the current file being processed
  print(file)
  
  # Get a list of files for the current capture and sample
  files_list <- list.files(pattern = file)
  print(files_list)
  
  # Specify the files you want to process for each sample
  files <- c(
    paste0(file, ".AllCDS.all.split.txt"),
    paste0(file, ".AllExon.all.split.txt"),
    paste0(file, ".AllGene.all.split.txt"),
    paste0(file, ".AllUTR.all.split.txt")
  )
  print(files)
  
  # Labels for different data types
  labs <- c("CDS", "Exon", "Gene", "UTR")
  
  # Initialize a list to store data for each data type
  cov <- list()
  
  # Loop through each file and data type
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[i]][, 2])
    cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample <- labs[i]
  }
  
  # Combine data for all data types
  cov_df <- do.call("rbind", cov)
  names(cov_df)[1:2] <- c("depth", "fraction")
  
  # Color palette
  pcbPalette <- c("#009E73", "#D55E00", "#CC79A7", "#56B4E9")
  
  # Create a ggplot
  p1 <- ggplot(cov_df, aes(x = depth, y = cov_cumul, color = sample, linetype = sample)) +
    xlim(0, 10) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    scale_color_manual(values = pcbPalette) +
    scale_fill_manual(values = pcbPalette) +
    ggtitle(paste("Read Depth Distribution for", capture, "(", sample, ")")) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(axis.text = element_text(size = 5),axis.title = element_text(size = 5,face = "bold")) +
    theme(plot.title = element_text(size = 5, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")
  
  # Append the plot to the list
  plot_list[[sample]] <- p1
  
  # Save the individual plot as a PNG file
  png(
    filename = paste0("Figure2_", capture, "_", sample, ".png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
}

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(
  grobs = plot_list,
  ncol = 3,  # Adjust the number of columns as needed
  top = textGrob(paste("Read Depth Distribution for", capture, "across Samples"))
)

# Save the combined plot as a PNG file
ggsave(file = paste0("Figure2_", capture, "_Combined.png"), pcombined)
```

```{r Create one graph showing the distribution of CDS, Exon, Gene, and UTR for Capture1 across all samples}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define a list of samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Initialize an empty list to store ggplot objects
plot_list <- list()

# Define captures for Capture1
capture <- "capture1"
files <- paste0(capture, "_", samples)
labs <- c("CDS", "Exon", "Gene", "UTR")
cov <- list()

for (i in 1:length(files)) {
  files_list <- list.files(pattern = files[i])
  print(files_list)
  current_files <- c(paste0(files[i], ".AllCDS.all.split.txt"), paste0(files[i], ".AllExon.all.split.txt"), paste0(files[i], ".AllGene.all.split.txt"), paste0(files[i], ".AllUTR.all.split.txt"))
  print(current_files)
  
  for (j in 1:length(current_files)) {
    cov[[j]] <- read.table(current_files[j])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[j]][, 2])
    cov[[j]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[j]]$sample <- paste(samples[i], labs[j], sep = "_")
  }
}

cov_df <- do.call("rbind", cov)
names(cov_df)[1:2] <- c("depth", "fraction")
pcbPalette <- c("#009E73", "#D55E00", "#CC79A7", "#56B4E9")

p1 <- ggplot(cov_df, aes(x = depth, y = cov_cumul, color = sample, fill = sample)) + xlim(0, 5) +
  scale_alpha(guide = 'none') +
  geom_line(size = 1.5) +
  scale_color_manual(values = pcbPalette) +
  scale_fill_manual(values = pcbPalette) +
  ggtitle(paste("Read Depth Distribution for", capture, "across Samples")) +
  ylab("% of Bases > Depth") +
  xlab("Depth") +
  theme_bw() +
  theme(plot.title = element_text(size = 12, face = "bold", hjust = 0.5)) +
  theme(legend.position = "right")

# Store the combined plot in the list
plot_list[[capture]] <- p1

# Create a single PNG file with the combined graph for Capture1
png(filename = "Figure2_Capture1_Distribution_Across_Samples.png", type = "cairo", units = "px", width = 5600, 
    height = 3000, res = 600, bg = "transparent")

# Plot and save the combined graph
print(p1)

dev.off()
```

```{r For Capture 1 add an additional column to combined_cov_df that identifies the unique sample}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the capture you want to isolate
capture <- "capture1"

# Initialize an empty list to store ggplot objects
plot_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Initialize a list to store data for each sample
cov_list <- list()

# Loop through each sample
for (sample in samples) {
  # Create the file name based on the sample and capture
  file <- paste0(capture, "_", sample)
  
  # Print the current file being processed
  print(file)
  
  # Get a list of files for the current capture and sample
  files_list <- list.files(pattern = file)
  print(files_list)
  
  # Specify the files you want to process for each sample
  files <- c(
    paste0(file, ".AllCDS.all.split.txt"),
    paste0(file, ".AllExon.all.split.txt"),
    paste0(file, ".AllGene.all.split.txt"),
    paste0(file, ".AllUTR.all.split.txt")
  )
  print(files)
  
  # Labels for different data types
  labs <- c("CDS", "Exon", "Gene", "UTR")
  
  # Initialize a list to store data for each data type
  cov <- list()
  
  # Loop through each file and data type
  for (i in 1:length(files)) {
    cov[[i]] <- read.table(files[i])[, c(2, 5)]
    cov_cumul <- 1 - cumsum(cov[[i]][, 2])
    cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
    cov[[i]]$sample <- labs[i]
    cov[[i]]$unique_sample <- paste(sample, labs[i], sep = "_")  # Add unique sample identifier
  }
  
  # Combine data for all data types for the current sample
  cov_df <- do.call("rbind", cov)
  names(cov_df)[1:2] <- c("depth", "fraction")
  
  # Store the combined data for the current sample
  cov_list[[sample]] <- cov_df
}

# Combine data from all samples
combined_cov_df <- do.call("rbind", cov_list)

# Get unique sample names
unique_samples <- unique(combined_cov_df$unique_sample)

# Generate a color palette with unique colors for each sample
color_palette <- rainbow(length(unique_samples))

# Create a named vector to map sample names to colors
sample_colors <- setNames(color_palette, unique_samples)

# Create a ggplot for the combined data
p1 <- ggplot(combined_cov_df, aes(x = depth, y = cov_cumul, color = unique_sample, linetype = sample)) +
  xlim(0, 5) +
  scale_alpha(guide = 'none') +
  geom_line(size = 0.5) +
  scale_color_manual(values = sample_colors) +  # Use the sample_colors palette
  scale_fill_manual(values = sample_colors) +  # Use the sample_colors palette
  ggtitle(paste("Read Depth Distribution for", capture, "across Samples")) +
  ylab("% of Bases > Depth") +
  xlab("Depth") +
  theme_bw() +
  theme(plot.title = element_text(size = 8, face = "bold", hjust = 0.5))

# Save the combined plot as a PNG file
png(
  filename = paste0("Figure2_", capture, "_Combined.png"),
  type = "cairo",
  units = "px",
  width = 5600,
  height = 3500,
  res = 600,
  bg = "transparent"
)

print(p1)
dev.off()

```

```{r Expand previous to each Capture}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the list of captures
captures <- c("capture1", "capture4", "capture7", "capture10", "capture13", "capture16")

# Initialize an empty list to store ggplot objects for each capture
plot_list <- list()

# Initialize lists to store average and variance data
average_list <- list()
variance_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Loop through each capture
for (capture in captures) {
  
  # Initialize a list to store data for each sample within the capture
  cov_list <- list()
  
  # Loop through each sample
  for (sample in samples) {
    # Create the file name based on the sample and capture
    file <- paste0(capture, "_", sample)
    
    # Print the current file being processed
    print(file)
    
    # Get a list of files for the current capture and sample
    files_list <- list.files(pattern = file)
    print(files_list)
    
    # Specify the files you want to process for each sample
    files <- c(
      paste0(file, ".AllCDS.all.split.txt"),
      paste0(file, ".AllExon.all.split.txt"),
      paste0(file, ".AllGene.all.split.txt"),
      paste0(file, ".AllUTR.all.split.txt")
    )
    print(files)
    
    # Labels for different data types
    labs <- c("CDS", "Exon", "Gene", "UTR")
    
    # Initialize a list to store data for each data type
    cov <- list()
    
    # Loop through each file and data type
    for (i in 1:length(files)) {
      cov[[i]] <- read.table(files[i])[, c(2, 5)]
      cov_cumul <- 1 - cumsum(cov[[i]][, 2])
      cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
      cov[[i]]$sample <- labs[i]
      cov[[i]]$unique_sample <- paste(sample, labs[i], sep = "_")  # Add unique sample identifier
    }
    
    # Combine data for all data types for the current sample
    cov_df <- do.call("rbind", cov)
    names(cov_df)[1:2] <- c("depth", "fraction")
    
    # Store the combined data for the current sample
    cov_list[[sample]] <- cov_df
  }
  
  # Combine data from all samples within the capture
  combined_cov_df <- do.call("rbind", cov_list)
  
  # Get unique sample names
  unique_samples <- unique(combined_cov_df$unique_sample)
  
  # Generate a color palette with unique colors for each sample
  color_palette <- rainbow(length(unique_samples))
  
  # Create a named vector to map sample names to colors
  sample_colors <- setNames(color_palette, unique_samples)
  
  # Create a ggplot for the combined data for the current capture
  p1 <- ggplot(combined_cov_df, aes(x = depth, y = cov_cumul, color = unique_sample, linetype = sample)) +
    xlim(0, 5) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    scale_color_manual(values = sample_colors) +
    ggtitle(paste("Read Depth Distribution for", capture)) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(plot.title = element_text(size = 6, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")  # Remove the legend
  
  # Save the combined plot as a PNG file for the current capture
  png(
    filename = paste0("Figure2_", capture, "_Combined_stitch.png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
  
  # Store the combined plot in the list
  plot_list[[capture]] <- p1
}

legend_p1 <- get_legend(p1 + theme(legend.position = "right"))

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, legend_p1, ncol = 3, top = textGrob("Read Depth Distribution across Captures"))

# Save the combined plot as a PNG file for all captures
ggsave(file = "Figure2_All_Captures_Combined.png", pcombined)
```

```{r Calculate the average and variance for each capture across all samples and data types}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the list of captures
captures <- c("capture1", "capture4", "capture7", "capture10", "capture13", "capture16")

# Initialize an empty list to store ggplot objects for each capture
plot_list <- list()

# Initialize lists to store average and variance data
average_list <- list()
variance_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Loop through each capture
for (capture in captures) {
  
  # Initialize a list to store data for each sample within the capture
  cov_list <- list()
  
  # Loop through each sample
  for (sample in samples) {
    # Create the file name based on the sample and capture
    file <- paste0(capture, "_", sample)
    
    # Print the current file being processed
    print(file)
    
    # Get a list of files for the current capture and sample
    files_list <- list.files(pattern = file)
    print(files_list)
    
    # Specify the files you want to process for each sample
    files <- c(
      paste0(file, ".AllCDS.all.split.txt"),
      paste0(file, ".AllExon.all.split.txt"),
      paste0(file, ".AllGene.all.split.txt"),
      paste0(file, ".AllUTR.all.split.txt")
    )
    print(files)
    
    # Labels for different data types
    labs <- c("CDS", "Exon", "Gene", "UTR")
    
    # Initialize a list to store data for each data type
    cov <- list()
    
    # Loop through each file and data type
    for (i in 1:length(files)) {
      cov[[i]] <- read.table(files[i])[, c(2, 5)]
      cov_cumul <- 1 - cumsum(cov[[i]][, 2])
      cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
      cov[[i]]$sample <- labs[i]
      cov[[i]]$unique_sample <- paste(sample, labs[i], sep = "_")  # Add unique sample identifier
    }
    
    # Combine data for all data types for the current sample
    cov_df <- do.call("rbind", cov)
    names(cov_df)[1:2] <- c("depth", "fraction")
    
    # Store the combined data for the current sample
    cov_list[[sample]] <- cov_df
  }
  
  # Combine data from all samples within the capture
  combined_cov_df <- do.call("rbind", cov_list)
  
  # Calculate the average and variance for each depth value across all samples and data types
  average_df <- combined_cov_df %>%
    group_by(depth) %>%
    summarize(average = mean(cov_cumul), variance = var(cov_cumul))
  
  # Store the average and variance data in lists
  average_list[[capture]] <- average_df
  variance_list[[capture]] <- average_df
  
  # Get unique sample names
  unique_samples <- unique(combined_cov_df$unique_sample)
  
  # Generate a color palette with unique colors for each sample
  color_palette <- rainbow(length(unique_samples))
  
  # Create a named vector to map sample names to colors
  sample_colors <- setNames(color_palette, unique_samples)
  
  # Create a ggplot for the combined data for the current capture
  p1 <- ggplot(combined_cov_df, aes(x = depth, y = cov_cumul, color = unique_sample, linetype = sample)) +
    xlim(0, 5) +
    scale_alpha(guide = 'none') +
    geom_line(size = 1.5) +
    scale_color_manual(values = sample_colors) +
    ggtitle(paste("Read Depth Distribution for", capture)) +
    ylab("% of Bases > Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(plot.title = element_text(size = 6, face = "bold", hjust = 0.5)) +
    theme(legend.position = "none")  # Remove the legend
  
  # Save the combined plot as a PNG file for the current capture
  png(
    filename = paste0("Figure2_", capture, "_Combined.png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p1)
  dev.off()
  
  # Store the combined plot in the list
  plot_list[[capture]] <- p1
}

# Combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 3, top = textGrob("Read Depth Distribution across Captures"))

# Save the combined plot as a PNG file for all captures
ggsave(file = "Figure2_All_Captures_Combined_Avg_Var.png", pcombined)
```


```{r Create separate plots for plotting the average and variance for each capture with depth on the x-axis and the average depth on the y-axis}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the list of captures
captures <- c("capture1", "capture4", "capture7", "capture10", "capture13", "capture16")

# Initialize lists to store average and variance data
average_list <- list()
variance_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Loop through each capture
for (capture in captures) {
  
  # Initialize a list to store data for each sample within the capture
  cov_list <- list()
  
  # Loop through each sample
  for (sample in samples) {
    # Create the file name based on the sample and capture
    file <- paste0(capture, "_", sample)
    
    # Get a list of files for the current capture and sample
    files_list <- list.files(pattern = file)
    
    # Specify the files you want to process for each sample
    files <- c(
      paste0(file, ".AllCDS.all.split.txt"),
      paste0(file, ".AllExon.all.split.txt"),
      paste0(file, ".AllGene.all.split.txt"),
      paste0(file, ".AllUTR.all.split.txt")
    )
    
    # Labels for different data types
    labs <- c("CDS", "Exon", "Gene", "UTR")
    
    # Initialize a list to store data for each data type
    cov <- list()
    
    # Loop through each file and data type
    for (i in 1:length(files)) {
      cov[[i]] <- read.table(files[i])[, c(2, 5)]
      cov_cumul <- 1 - cumsum(cov[[i]][, 2])
      cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
      cov[[i]]$sample <- labs[i]
      cov[[i]]$unique_sample <- paste(sample, labs[i], sep = "_")  # Add unique sample identifier
    }
    
    # Combine data for all data types for the current sample
    cov_df <- do.call("rbind", cov)
    names(cov_df)[1:2] <- c("depth", "fraction")
    
    # Store the combined data for the current sample
    cov_list[[sample]] <- cov_df
  }
  
  # Combine data from all samples within the capture
  combined_cov_df <- do.call("rbind", cov_list)
  
  # Calculate the average and variance for each depth value across all samples and data types
  average_df <- combined_cov_df %>%
    dplyr::group_by(depth) %>%
    dplyr::summarize(average = mean(cov_cumul))
  
  variance_df <- combined_cov_df %>%
    dplyr::group_by(depth) %>%
    dplyr::summarize(variance = var(cov_cumul))
  
  # Store the average and variance data in lists
  average_list[[capture]] <- average_df
  variance_list[[capture]] <- variance_df
}

# Create separate plots for average and variance for each capture
for (i in 1:length(captures)) {
  capture <- captures[i]
  
  # Plot average
  p_average <- ggplot(average_list[[i]], aes(x = depth, y = average)) +
    geom_line() +
    xlim(0,5) +
    #ylim(0,0.5) +
    ggtitle(paste("Average Depth for", capture, "across Samples")) +
    ylab("Average Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(plot.title = element_text(size = 6, face = "bold", hjust = 0.5))
  
  # Plot variance
  p_variance <- ggplot(variance_list[[i]], aes(x = depth, y = variance)) +
    geom_line() +
    xlim(1,5) +
    ggtitle(paste("Variance of Depth for", capture, "across Samples")) +
    ylab("Variance") +
    xlab("Depth") +
    theme_bw() +
    theme(plot.title = element_text(size = 6, face = "bold", hjust = 0.5))
  
  # Save the plots as PNG files
  png(
    filename = paste0("Figure2_", capture, "_Average.png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  print(p_average)
  dev.off()
  
  png(
    filename = paste0("Figure2_", capture, "_Variance.png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  print(p_variance)
  dev.off()
}
```


```{r Plotting Captures as average and variance}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the list of captures
captures <- c("capture1", "capture4", "capture7", "capture10", "capture13", "capture16")

# Initialize lists to store average and variance data
average_list <- list()
variance_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Initialize an empty list to store ggplot objects for each capture
plot_list <- list()

# Loop through each capture
for (capture in captures) {
  
  # Initialize a list to store data for each sample within the capture
  cov_list <- list()
  
  # Loop through each sample
  for (sample in samples) {
    # Create the file name based on the sample and capture
    file <- paste0(capture, "_", sample)
    
    # Get a list of files for the current capture and sample
    files_list <- list.files(pattern = file)
    
    # Specify the files you want to process for each sample
    files <- c(
      paste0(file, ".AllCDS.all.split.txt"),
      paste0(file, ".AllExon.all.split.txt"),
      paste0(file, ".AllGene.all.split.txt"),
      paste0(file, ".AllUTR.all.split.txt")
    )
    
    # Labels for different data types
    labs <- c("CDS", "Exon", "Gene", "UTR")
    
    # Initialize a list to store data for each data type
    cov <- list()
    
    # Loop through each file and data type
    for (i in 1:length(files)) {
      cov[[i]] <- read.table(files[i])[, c(2, 5)]
      cov_cumul <- 1 - cumsum(cov[[i]][, 2])
      cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
      cov[[i]]$sample <- labs[i]
      cov[[i]]$unique_sample <- paste(sample, labs[i], sep = "_")  # Add unique sample identifier
    }
    
    # Combine data for all data types for the current sample
    cov_df <- do.call("rbind", cov)
    names(cov_df)[1:2] <- c("depth", "fraction")
    
    # Store the combined data for the current sample
    cov_list[[sample]] <- cov_df
  }
  
  # Combine data from all samples within the capture
  combined_cov_df <- do.call("rbind", cov_list)
  
  average_df <- combined_cov_df %>%
    dplyr::group_by(depth) %>%
    dplyr::summarize(average = mean(cov_cumul))
  
  # Store the average and variance data in lists
  average_list[[capture]] <- average_df
}

# Create separate plots for average and variance for each capture
for (i in 1:length(captures)) {
  capture <- captures[i]
  
  # Plot average
  p_average <- ggplot(average_list[[i]], aes(x = depth, y = average)) +
    xlim(0,10) +
    geom_line() +
    ggtitle(paste("Average Depth for", capture)) +
    ylab("Average Depth") +
    xlab("Depth") +
    theme_bw() +
    theme(plot.title = element_text(size = 8, face = "bold", hjust = 0.5))
  
  # Save the plots as PNG files
  png(
    filename = paste0("Figure2_", capture, "_Average.png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  print(p_average)
  dev.off()
  plot_list[[capture]] <- p_average
}

pcombined <- grid.arrange(grobs = plot_list, ncol = 2, top = textGrob("Average Depth Distribution across Captures"))

# Save the combined plot as a PNG file for all captures
ggsave(file = "Figure2_All_Capture_Avg_Combined.png", pcombined)
```



```{r Average %Bases > depth combined}
# Set your working directory
setwd("/home/jgreen/eager_obj1b/04_coverage_analysis/01_genome_region/01_merged/")

# Define the list of captures
captures <- c("capture1", "capture4", "capture7", "capture10", "capture13", "capture16")

# Initialize lists to store average and variance data
average_list <- list()
variance_list <- list()

# List of all samples
samples <- c("b1", "b2", "g1", "g7", "k1", "k3", "m5", "m7", "n1", "n2")

# Initialize an empty list to store ggplot objects for each capture
plot_list <- list()

# Loop through each capture
for (capture in captures) {
  
  # Initialize a list to store data for each sample within the capture
  cov_list <- list()
  
  # Loop through each sample
  for (sample in samples) {
    # Create the file name based on the sample and capture
    file <- paste0(capture, "_", sample)
    
    # Get a list of files for the current capture and sample
    files_list <- list.files(pattern = file)
    
    # Specify the files you want to process for each sample
    files <- c(
      paste0(file, ".AllCDS.all.split.txt"),
      paste0(file, ".AllExon.all.split.txt"),
      paste0(file, ".AllGene.all.split.txt"),
      paste0(file, ".AllUTR.all.split.txt")
    )
    
    # Labels for different data types
    labs <- c("CDS", "Exon", "Gene", "UTR")
    
    # Initialize a list to store data for each data type
    cov <- list()
    
    # Loop through each file and data type
    for (i in 1:length(files)) {
      cov[[i]] <- read.table(files[i])[, c(2, 5)]
      cov_cumul <- 1 - cumsum(cov[[i]][, 2])
      cov[[i]]$cov_cumul <- c(1, cov_cumul[-length(cov_cumul)])
      cov[[i]]$sample <- labs[i]
      cov[[i]]$unique_sample <- paste(sample, labs[i], sep = "_")  # Add unique sample identifier
    }
    
    # Combine data for all data types for the current sample
    cov_df <- do.call("rbind", cov)
    names(cov_df)[1:2] <- c("depth", "fraction")
    
    # Store the combined data for the current sample
    cov_list[[sample]] <- cov_df
  }
  
  # Combine data from all samples within the capture
  combined_cov_df <- do.call("rbind", cov_list)
  
  # Calculate the average for each data type (CDS, Exon, Gene, UTR)
  average_df <- combined_cov_df %>%
    dplyr::group_by(sample, depth, sample_type = factor(sample, levels = c("CDS","Exon","Gene","UTR"))) %>%
    dplyr::summarize(
      average = mean(cov_cumul),
      Q1 = quantile(cov_cumul, 0.25),
      Q3 = quantile(cov_cumul, 0.75),
      variance = var(cov_cumul)
    )
  
  # Create a ggplot for the combined data for the current capture
   p <- ggplot(average_df, aes(x = depth, y = average, color = sample_type)) +
    scale_x_continuous(breaks = seq(0, 8, by = 1), limits = c(0, 8)) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    geom_ribbon(aes(ymin = Q1, ymax = Q3), alpha = 0.2) +
    ggtitle(paste(capture)) +
    ylab("Average %bases > depth") +
    xlab("Depth") +
    theme_bw() +
    theme(
      axis.text = element_text(size = 5),
      axis.title = element_text(size = 5, face = "bold"),
      plot.title = element_text(size = 5, face = "bold", hjust = 0.5),
      legend.position = "none"
    )
  
  # Save the combined plot as a PNG file for the current capture
  png(
    filename = paste0("Figure2_", capture, "_AverageBases_Depth.png"),
    type = "cairo",
    units = "px",
    width = 5600,
    height = 3000,
    res = 600,
    bg = "transparent"
  )
  
  print(p)
  dev.off()
  
  # Store the combined plot in the list
  plot_list[[capture]] <- p
}

plot1_legend <- ggplot(average_df, aes(x = depth, y = average, color = sample_type, fill = sample_type)) +
    scale_x_continuous(breaks = seq(0, 8, by = 1), limits = c(0, 8)) +
    scale_alpha(guide = 'none') +
    geom_line(size = 0.5) +
    geom_ribbon(aes(ymin = Q1, ymax = Q3), alpha = 0.2) +
    ggtitle(paste(capture)) +
    ylab("Average %bases > depth") +
    xlab("Depth") +
    theme_bw() +
    labs(fill="Region")
    theme(
      axis.text = element_text(size = 5),
      axis.title = element_text(size = 5, face = "bold"),
      plot.title = element_text(size = 5, face = "bold", hjust = 0.5),
      legend.position = "bottom")

plot1_legend

get_only_legend <- function(plot) { 
  plot_table <- ggplot_gtable(ggplot_build(plot)) 
  legend_plot <- which(sapply(plot_table$grobs, function(x) x$name) == "guide-box") 
  legend <- plot_table$grobs[[legend_plot]] 
  return(legend) 
} 

# extract legend from plot1 using above function 
legend <- get_only_legend(plot1_legend)

legend_p <- get_legend(p+theme(legend.position = "bottom",
                               legend.title = element_text(size = 8),
                               legend.text = element_text(size=6))) 

# combine all the ggplot objects in the list using grid.arrange()
pcombined <- grid.arrange(grobs = plot_list, ncol = 3, top = textGrob("Average Depth across Captures"))
pcombined_legend <- plot_grid(pcombined, legend_p, ncol=1, rel_heights = c(.5, .1))

# Save the combined plot as a PNG file for all captures
ggsave(file = "Figure2_All_Captures_Average_Depth.png", pcombined)
ggsave(file = "Figure2_All_Captures_Average_Depth_Legend.png", pcombined_legend)

```